### DISCUSSION TITLE
Creating a Linux app with Tray Icon only

### DISCUSSION THREAD

**Discussion by aleDsz** [raysan5/raylib#3791] [Q&A]
Hi, I've plans (on my own) to allow [Livebook Desktop App](https://github.com/livebook-dev/livebook) to be built for Linux distros and I've looking into some GUI library projects to create a single binary for many distros it can run, but it's kinda hard to have a GUI that works everywhere (only Linux in this case).

This is an example of how we works in MacOS, which is only the system tray icon with a menu and all these actions interacts with an Elixir API.  And we already have a .NET implementation for Windows.

![image](https://github.com/raysan5/raylib/assets/6402997/6e4864e5-54ee-4899-95e3-45ce0ef633a2)

That said, is there a way to create a Linux app with tray icon + menu? If yes, a tiny example would help :)

Also, I'm open to any suggestion.

**Comment by DesLandysh** [comment]
I guess it would depended on what kind window manager you will use. 
that's example for Gnome: [AppIndicator](https://extensions.gnome.org/extension/615/appindicator-support/), it uses gnome-extensions, maybe also you should get libs for GTK-3 or GTK-4
for Mate, LFCE or KDE - the most popular WMs - it definitely has own solutions, also you should know that these ones uses X.org/X-system protocol, but Gnome uses Wayland protocol, however for mate/lfce/kde it should be easier:
that's for KDE: [PlasmaSystemTray widget](https://zren.github.io/2018/11/17/exploring-plasmas-systray-widget)

**Comment by gen2brain** [comment]
@aleDsz Nowadays old X system tray protocol is not used. You can use D-Bus/libdbus and StatusNotifierItem to set the icon, see https://www.freedesktop.org/wiki/Specifications/StatusNotifierItem/. The problem is, for the menu, you will need some toolkit, like GTK or Qt, and that means you must somehow integrate into their main loop.
